### PR TITLE
vm-monitor: Remove support for file cache in tmpfs

### PR DESCRIPTION
**Pull request by sharnoff**
ref neondatabase/cloud#7516.

We switched everything over to file cache on disk, now time to remove support for having it in tmpfs.

Out of an abundance of caution, this PR should not be merged until the control plane changes guaranteeing no tmpfs been released (neondatabase/cloud#7522) — leaving this as a draft PR until then.

Releasing this PR unblocks merging neondatabase/cloud#7523.

### CODE REVIEW

**Comment by sharnoff** [comment]
Marked this no longer draft. Control plane changes have been merged but not yet released.

File: compute_tools/src/bin/compute_ctl.rs [hunk 1]
```diff
@@ -482,6 +480,8 @@ fn cli() -> clap::Command {
                 .value_name("FILECACHE_CONNSTR"),
         )
         .arg(
+            // DEPRECATED, NO LONGER DOES ANYTHING.
+            // See https://github.com/neondatabase/cloud/issues/7516
             Arg::new("file-cache-on-disk")
                 .long("file-cache-on-disk")
                 .action(clap::ArgAction::SetTrue),
```
Comment (koivunej) [inline compute_tools/src/bin/compute_ctl.rs:487]: If this could be used in some environment, could log a warning when true.

**Review by koivunej — APPROVED** [review]
Looks good to me.

File: compute_tools/src/bin/compute_ctl.rs (hunk 1 above)
Comment (sharnoff) [reply compute_tools/src/bin/compute_ctl.rs:487]: Hm, yeah I was just thinking of this. Curious for your thoughts:

Currently the control plane always sets this argument (neondatabase/cloud#7522), then once this PR is released, we'll change the control plane to never set this argument  (neondatabase/cloud#7523).

So, _right now_ it'd be good to have warnings for when this arg is _not_ set. But we'll end up with warnings once we make the control plane change anyways, so... I don't know.

Comment (koivunej) [reply compute_tools/src/bin/compute_ctl.rs:487]: Aah well, if it's always set, then perhaps not. I was thinkin gif some tests shouldn't use this flag anymore, or if there were more conditions to using it. Feel free to close.

**Pull request merged**